### PR TITLE
Replace 2d constant array indexing with select4

### PR DESCRIPTION
**Pull request by eh-unity**
### Checklist for PR maker
- [x] Have you updated the changelog? Each package has a `CHANGELOG.md` file.
---
### Purpose of this PR
Why is this PR needed, what hard problem is it solving/fixing?

---
### Testing status
Improve lighting shader code gen / perf.

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

**Review by pbbastian — APPROVED** [review]
LGTM! Great to finally get this change in :D

We should verify this with some performance testing.

File: com.unity.render-pipelines.universal/ShaderLibrary/ShaderVariablesFunctions.hlsl [hunk 1]
```diff
@@ -346,6 +346,25 @@ float2 GetNormalizedScreenSpaceUV(float4 positionCS)
     return GetNormalizedScreenSpaceUV(positionCS.xy);
 }
 
+// Select uint4 component by index.
+// Helper to improve codegen for 2d indexing (data[x][y])
+// Replace:
+// data[i / 4][i % 4];
+// with:
+// select4(data[i / 4], i % 4);
+uint Select4(uint4 v, uint i)
```
Comment (phi-lira) [inline com.unity.render-pipelines.universal/ShaderLibrary/ShaderVariablesFunctions.hlsl:355]: Does this code compile for GLES2? If not you can ifdef for SHADER_API_GLES2

**Review by phi-lira — APPROVED** [review]

File: com.unity.render-pipelines.universal/ShaderLibrary/ShaderVariablesFunctions.hlsl (hunk 1 above)
Comment (eh-unity) [reply com.unity.render-pipelines.universal/ShaderLibrary/ShaderVariablesFunctions.hlsl:355]: It should. I tried building for webGL to be sure. It seems to build without errors.

**Review by simon-engelbrecht-soerensen — APPROVED** [review]

**Comment by phi-lira** [comment]
Closing but keeping the branch open. Please delete branch if you don't plan to reopen this PR.

**Pull request closed without merging**